assume/reinforcement_learning/network_architecture.py

```diff
@@ -0,0 +1,188 @@
+        super().__init__()
+
+
+class MLPActor(Actor):
+    """
+    The neurnal network for the MLP actor.
+    """
+
+    def __init__(self, obs_dim: int, act_dim: int, float_type):
+        super().__init__()
+
+        self.FC1 = nn.Linear(obs_dim, 256, dtype=float_type)
+        self.FC2 = nn.Linear(256, 128, dtype=float_type)
+        self.FC3 = nn.Linear(128, act_dim, dtype=float_type)
+
+    def forward(self, obs):
+        x = F.relu(self.FC1(obs))
+        x = F.relu(self.FC2(x))
+        x = F.softsign(self.FC3(x))
+        # x = th.tanh(self.FC3(x))
+
+        return x
+
+
+class LSTMActor(nn.Module):
+    """
+    The LSTM recurrent neurnal network for the actor.
+
+    Based on: "Multi-Period and Multi-Spatial Equilibrium Analysis in Imperfect Electricity Markets"
+    by Ye at al. (2019)
+    """
+
+    def __init__(self, obs_dim: int, act_dim: int, float_type):
+        super().__init__()
+        self.float_type = float_type
+        self.no_of_timeseries = 2  # TODO: variable no. of input timeseries
+        self.forecast_horizon = int(
+            (obs_dim - 2) / self.no_of_timeseries
+        )  # TODO: variable number of additional input features, currently 2 (marginal costs, capacity)
+
+        self.LSTM1 = nn.LSTMCell(self.no_of_timeseries, 8, dtype=float_type)
+        self.LSTM2 = nn.LSTMCell(8, 16, dtype=float_type)
+
+        # input size defined by forecast horizon and concatenated with capacity and marginal cost values
+        self.FC1 = nn.Linear(self.forecast_horizon * 16 + 2, 128, dtype=float_type)
+        self.FC2 = nn.Linear(128, act_dim, dtype=float_type)
+
+    def forward(self, obs):
+        if obs.dim() not in (1, 2):
+            raise ValueError(
+                f"LSTMCell: Expected input to be 1D or 2D, got {obs.dim()}D instead"
+            )
+
+        is_batched = obs.dim() == 2
+        if not is_batched:
+            obs = obs.unsqueeze(0)
+
+        x1, x2 = obs.split(
+            [obs.shape[1] - 2, 2], dim=1
+        )  # TODO: variable no. of mc and capacity values, currently 2
+        x1 = x1.reshape(
+            -1, self.no_of_timeseries, int(x1.shape[1] / self.no_of_timeseries)
+        )  # Shape: [no_of_timeseries, forecast_horizon] TODO: error handling for division
+
+        h_t = th.zeros(x1.size(0), 8, dtype=self.float_type, device=obs.device)
+        c_t = th.zeros(x1.size(0), 8, dtype=self.float_type, device=obs.device)
+
+        h_t2 = th.zeros(x1.size(0), 16, dtype=self.float_type, device=obs.device)
+        c_t2 = th.zeros(x1.size(0), 16, dtype=self.float_type, device=obs.device)
+
+        outputs = []
+
+        for time_step in x1.split(1, dim=2):
+            time_step = time_step.reshape(-1, 2)
+            h_t, c_t = self.LSTM1(time_step, (h_t, c_t))
+            h_t2, c_t2 = self.LSTM2(h_t, (h_t2, c_t2))
+            outputs += [h_t2]
+
+        outputs = th.cat(outputs, dim=1)
+        x = th.cat((outputs, x2), dim=1)
+
+        x = F.relu(self.FC1(x))
```

Comment:
did you try to use here after the FC1 a dropot of like 0.2 maybe? just wondering, not for this PR. We could test it out with the mlp and this one if this would improve the generalization

Reply:
that's currently in our backlog, we haven't tried it yet. but we will keep it in mind!